modules/github_integration/app/models/deploy_target.rb

```diff
@@ -0,0 +1,40 @@
+#-- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) 2012-2024 the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+
+class DeployTarget < ApplicationRecord
+  self.inheritance_column = nil
+
+  store_accessor :options, :api_key
+
+  has_many :deploy_status_checks, dependent: :destroy
+
+  validates_presence_of :host
```

Comment:
Since we have a [unique index here](https://github.com/opf/openproject/pull/15451/files#diff-19c6210cb5d9070ea7e81240ef4088c0544d4eda3b34e24cec25002504e5f880R11), should we also have a uniqueness validator?

```suggestion
  validates_presence_of :host
  validates_uniqueness_of :host
```

Reply:
In a future version probably we should validate the host format, ie: to not allow the protocol to be present.